File: server/src/main/java/com/smartexpense/api/domain/model/Notification.java
package com.smartexpense.api.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {
    private UUID id;
    private UUID userId;
    private String type; // 'BUDGET_WARNING', 'AI_TIP', etc.
    private String content;
    private Boolean isRead;
    private LocalDateTime createdAt;
}

File: server/bin/main/com/smartexpense/api/domain/model/Notification.kt
package com.smartexpense.api.domain.model

import java.time.LocalDateTime
import java.util.UUID

data class Notification(
    val id: UUID? = null,
    val userId: UUID,
    val type: String,
    val content: String,
    val isRead: Boolean = false,
    val createdAt: LocalDateTime? = null
)
